pkg/kubecli/install.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ * Copyright 2018 The Kubernetes Authors.
+ *
+ */
+
+package kubecli
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var (
+	kubectlPluginPath string
+)
+
+// Plugin holds everything needed to register a
+// plugin as a command. Usually comes from a descriptor file.
+type Plugin struct {
+	Name      string  `json:"name"`
+	ShortDesc string  `json:"shortDesc"`
+	LongDesc  string  `json:"longDesc,omitempty"`
+	Example   string  `json:"example,omitempty"`
+	Command   string  `json:"command"`
+	Flags     []Flag  `json:"flags,omitempty"`
+	Tree      Plugins `json:"tree,omitempty"`
+}
+
+// Source holds the location of a given plugin in the filesystem.
+type Source struct {
+	Dir            string `json:"-"`
+	DescriptorName string `json:"-"`
+}
+
+// Plugins is a list of plugins.
+type Plugins []*Plugin
+
+// Flag describes a single flag supported by a given plugin.
+type Flag struct {
+	Name      string `json:"name"`
+	Shorthand string `json:"shorthand,omitempty"`
+	Desc      string `json:"desc"`
+	DefValue  string `json:"defValue,omitempty"`
+}
+
+func InstallVirtPlugin(cmd *cobra.Command) error {
+	err := getPluginFolder()
+	if err != nil {
+		return err
+	}
+
+	// Create virt folder
+	err = os.MkdirAll(kubectlPluginPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	plugin := MakePluginConfiguration(cmd)
+
+	err = writePluginYaml(plugin)
+	if err != nil {
+		return err
+	}
+
+	return copyVirtctlFile()
+}
+
+func getPluginFolder() error {
+	globalPluginPath := os.Getenv("KUBECTL_PLUGINS_PATH")
+	if len(globalPluginPath) > 0 {
```

Comment:
on other places, we use `!= ""`, why not here? Also, you can just use LookupEnv https://golang.org/pkg/os/#LookupEnv.

```
if globalPluginPath, defined := os.LookupEnv("KUBECTL_PLUGINS_PATH"); defined {
    ...
}
```